test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -108,3 +115,53 @@ session_simulator::timestamp_transaction_uint(const std::string &ts_type, uint64
 
     return (0);
 }
+
+int
+session_simulator::query_timestamp(
+  const std::string &config, std::string &hex_ts, bool &ts_supported)
+{
+    std::string query_timestamp;
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+
+    /* For an empty config default to read timestamp. */
+    if (config.empty())
+        query_timestamp = "read";
+    else {
+        std::map<std::string, std::string> config_map;
+
+        ts_manager->parse_config(config, config_map);
+
+        /* For query timestamp we only expect one config. */
+        if (config_map.size() != 1)
+            WT_SIM_RET_MSG(EINVAL, "Incorrect config (" + config + ") passed in query timestamp");
+
+        auto pos = config_map.find("get");
+        if (pos == config_map.end())
+            WT_SIM_RET_MSG(EINVAL, "Incorrect config (" + config + ") passed in query timestamp");
+
+        query_timestamp = pos->second;
+    }
+
+    ts_supported = false;
```

Comment:
Could we initialize it to "true" and set it to "false" in error case?
we generally try to follow the alphabetical order if we have multiple if's, so as per that the suggested order is {commit, first_commit, prepare,read}